src/circuit/gadget/ecc/chip/mul_fixed/base_field_elem.rs

```diff
@@ -0,0 +1,645 @@
+
+                    // Copy z_13_alpha_0_prime
+                    copy(
+                        &mut region,
+                        || "Copy z_13_alpha_0_prime",
+                        self.canon_advices[1],
+                        offset,
+                        &z_13_alpha_0_prime,
+                        perm,
+                    )?;
+
+                    // Copy z_13_alpha_0, which is α with the first 130 bits subtracted.
+                    copy(
+                        &mut region,
+                        || "Copy z_13_alpha_0",
+                        self.canon_advices[2],
+                        offset,
+                        &z_13_alpha_0,
+                        perm,
+                    )?;
+                }
+
+                Ok(())
+            },
+        )?;
+
+        Ok(result)
+    }
+
+    fn decompose_base_field_elem(
+        &self,
+        base_field_elem: CellValue<pallas::Base>,
+        offset: usize,
+        region: &mut Region<'_, pallas::Base>,
+    ) -> Result<EccBaseFieldElemFixed, Error> {
+        // Decompose base field element into 3-bit words.
+        let words: Vec<Option<u8>> = {
+            let words = base_field_elem.value().map(|base_field_elem| {
+                decompose_scalar_fixed::<pallas::Base>(
+                    base_field_elem,
+                    constants::L_ORCHARD_BASE,
+                    constants::FIXED_BASE_WINDOW_SIZE,
+                )
+            });
+
+            if let Some(words) = words {
+                words.into_iter().map(Some).collect()
+            } else {
+                vec![None; constants::NUM_WINDOWS]
+            }
+        };
+
+        // Initialize empty ArrayVec to store running sum values [z_1, ..., z_85].
+        let mut running_sum: ArrayVec<CellValue<pallas::Base>, { constants::NUM_WINDOWS }> =
+            ArrayVec::new();
+
+        // Assign running sum `z_i`, i = 0..=n, where z_{i+1} = (z_i - a_i) / (2^3)
+        // and `z_0` is initialized as `base_field_elem`.
+        let mut z = copy(
+            region,
+            || "z_0 = base_field_elem",
+            self.super_config.window,
+            offset,
+            &base_field_elem,
+            &self.super_config.perm,
+        )?;
+
+        for idx in 0..words.len() {
+            self.base_field_fixed_mul.enable(region, offset + idx)?;
+        }
+
+        let offset = offset + 1;
+
+        let eight_inv = pallas::Base::TWO_INV.square() * pallas::Base::TWO_INV;
+        for (idx, word) in words.iter().enumerate() {
+            // z_next = (z_cur - word) / (2^3)
+            let z_next = {
+                let word = word.map(|word| pallas::Base::from_u64(word as u64));
+                let z_next_val = z
+                    .value()
+                    .zip(word)
+                    .map(|(z_cur_val, word)| (z_cur_val - word) * eight_inv);
+                let cell = region.assign_advice(
+                    || format!("word {:?}", idx),
```

Comment:
This is assigning the running sum, not the words:
```suggestion
                    || format!("z_{:?}", idx + 1),
```